src/imap.rs

```diff
@@ -42,6 +41,10 @@ use crate::provider::Socket;
 use crate::scheduler::connectivity::ConnectivityStore;
 use crate::scheduler::InterruptInfo;
 use crate::stock_str;
+use crate::{
+    chat::{self, ChatId, ChatIdBlocked},
+    sql,
```

Comment:
I know this is probably rust-analyser doing this but could you split the `use crate::sql` out on a separate line?  Mixing these different styles of imports makes things really difficult to read to me because all imports from a particular module are no longer in the same place over time.  My general rule-of-thumb is to not ever use `::` inside a `::{ }` block.